plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -1653,6 +1657,21 @@ public void close()
             }
         }
 
+        private void ensureExtraBytesCapacity(int extraBytesCapacity)
+        {
+            int totalBytesCapacity = bufferSize + extraBytesCapacity;
+            if (buffer.length < totalBytesCapacity) {
+                // buffer length might be 0
+                int newBytesLength = max(buffer.length, INITIAL_BUFFER_SIZE);
+                if (totalBytesCapacity > newBytesLength) {
+                    // grow array by 50%
+                    newBytesLength = min(newBytesLength + (newBytesLength >> 1), partSize);
+                    newBytesLength = max(totalBytesCapacity, newBytesLength);
```

Comment:
actually this line has to be removed. Otherwise buffer might become larger than `partSize`

Reply:
I will try to simplify this method then